### DISCUSSION TITLE
Predefined or Flexible Types

### DISCUSSION THREAD

**Discussion by cahidenes** [bounswe/bounswe2023group2#56] [General]
All resources, events, actions, and needs have type attributes. Should these types be predefined, user-created, or both?

**Comment by cahidenes** [comment]
In my opinion, there should be predefined types, and these are shown to users while creating a new piece of information. But there should also be an option to create a new type.

**Comment by rburaksaritas** [comment]
As far as discussed in meetings and descriptions, I think **both** seems to be the better option.

**Comment by mehmetkuzu** [comment]
Predefined types may be the ones **given** by design.  
By example:
If we define a rescue operation and a rescue expert both as predefined types we define the related operation by design. (In the software) We should avoid this, I think. 

The predefined types are needs, resources, actions and events. (Which is fine)
But food, rescue expert, medicine should not be PTs. 
But for that we will need extra work. (Which is also fine)

**Comment by egecans** [comment]
I think these types should be defined both ways. 
As we discussed in the project owner's meeting, the app should be flexible. Therefore there will be cases that we couldn't consider before, so users can create things at that time. 
Another reason why this should be user-created is maybe users couldn't have enough time for selecting the most suitable options (especially while their charge&time are important in the disaster area), therefore they could enter things as flexibly as they can.
On the other hand, predefined type things could ease the user experience and labelling things. If users have enough time they should pick things' categories.

**Reply by mehmetkuzu** [reply]
Agree.
A convenient way: 
Some of the user defined types/operations are declared to the system by the system designer beforehand. They are not given by design but the system designers presee the possible uses.

**Comment by cahidenes** [comment]
Since no further ideas, I am closing this discussion. The agreed option is **both**.
